### PR TITLE
Add logic to allow users to "watch" a discussion. Fix non-passing tests

### PR DESCRIPTION
**Pull request by Arkitecht**
The purpose of this PR is to allow users to "watch" a discussion.

This is just built into the API at the moment, and could be added to the UI - as well as added to notifications to alert all watchers if a discussion is updated.

### CODE REVIEW

**Comment by AndreasElia** [comment]
Thank you for this @Arkitecht. I'll check it out hopefully tomorrow 😄

File: src/Traits/FireflyUser.php [hunk 1]
```diff
@@ -17,4 +18,20 @@ public function posts(): HasMany
...
+
+    public function isWatching(Discussion $discussion): bool
+    {
+        $watching = $this
+            ->watching()
+            ->select('discussion_id')
+            ->get()
+            ->pluck('discussion_id');
+
+        return $watching->contains($discussion->id);
```
Comment (tomirons) [inline src/Traits/FireflyUser.php:35]: Depending on where/how often this is used we can improve this.
```suggestion
        return $this
            ->watching()
            ->where('discussion_id', $discussion->id)
            ->exists();
```

File: tests/Feature/DiscussionTest.php [hunk 2]
```diff
@@ -17,7 +17,7 @@ public function test_discussion_was_created()
 
         $response = $this->actingAs($this->getUser())
             ->post('forum/g/example-group/d', [
-                'title' => 'Foo Bar',
+                'title'   => 'Foo Bar',
```
Comment (tomirons) [inline tests/Feature/DiscussionTest.php:20]: Please revert the formatting here.

File: tests/Feature/DiscussionTest.php [hunk 3]
```diff
@@ -26,7 +26,7 @@ public function test_discussion_was_created()
         $this->assertTrue($discussions->count() == 1);
         $this->assertDatabaseHas('discussions', [
             'title' => 'Foo Bar',
-            'slug' => 'foo-bar',
+            'slug'  => 'foo-bar',
```
Comment (tomirons) [inline tests/Feature/DiscussionTest.php:29]: Please revert the formatting here.

**Review by tomirons — CHANGES_REQUESTED** [review]
Well thank you for implementing this feature, it's a nice one to have! 👍 

I just have a few concerns that need to be addressed.

File: routes/web.php [hunk 4]
```diff
@@ -13,6 +13,8 @@
         Route::put('{discussion}-{slug}', 'DiscussionController@update')->name('discussion.update')->where(['discussion' => '[0-9]+']);
         Route::get('{discussion}-{slug}/edit', 'DiscussionController@edit')->name('discussion.edit')->where(['discussion' => '[0-9]+']);
         Route::delete('{discussion}-{slug}', 'DiscussionController@delete')->name('discussion.delete')->where(['discussion' => '[0-9]+']);
+        Route::put('{discussion}-{slug}/watch', 'DiscussionController@watch')->name('discussion.watch')->where(['discussion' => '[0-9]+']);
+        Route::put('{discussion}-{slug}/unwatch', 'DiscussionController@unwatch')->name('discussion.unwatch')->where(['discussion' => '[0-9]+']);
```
[the remark this answers is not in the document]
Comment (Arkitecht) [reply routes/web.php:17]: Sure, I was following the pin / unpin logic, but I can move these to a DiscussionWatchController - does that work?

Do you prefer using crud methods - ie. `store` for watch and `delete` for unwatch - or just use watch and unwatch as the method names?

[the remark this answers is not in the document]
Comment (tomirons) [reply routes/web.php:17]: > Sure, I was following the pin / unpin logic, but I can move these to a DiscussionWatchController - does that work?
Haha I didn't even realize this. We can move these to their own controllers in a follow up PR.

Let's stick to `store/delete` 👍

[the remark this answers is not in the document]
Comment (Arkitecht) [reply routes/web.php:17]: Will do.

I have the routes like this, to follow the crud conventions
```php
Route::post('{discussion}-{slug}/watch', 'DiscussionWatchController@store')->name('discussion.watch')->where(['discussion' => '[0-9]+']);
        Route::delete('{discussion}-{slug}/watch', 'DiscussionWatchController@delete')->name('discussion.unwatch')->where(['discussion' => '[0-9]+']);
```

That work?

File: src/Traits/FireflyUser.php (hunk 1 above)
Comment (Arkitecht) [reply src/Traits/FireflyUser.php]: Cleaner, thanks, I have updated the logic here.

File: src/Services/DiscussionServiceOrig.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Firefly\Services;
+
+use Firefly\Models\Discussion;
+use Firefly\Models\Group;
+use Illuminate\Http\Request;
+
+class DiscussionService
```
Comment (tomirons) [inline src/Services/DiscussionServiceOrig.php:9]: What is this file for?

**Review by tomirons — CHANGES_REQUESTED** [review]

File: src/Services/DiscussionServiceOrig.php (hunk 5 above)
Comment (Arkitecht) [reply src/Services/DiscussionServiceOrig.php]: Sorry, was using this to make sure I had restored all the formatting correctly, removed in the latest push

**Review by tomirons — APPROVED** [review]

**Pull request merged**